### PR TITLE
Setupresource

### PR DESCRIPTION
**Pull request by CaptBoscho**

### CODE REVIEW

File: java/src/shared/model/game/Game.java [hunk 1]
```diff
@@ -1169,8 +1195,8 @@ public Integer finishTurn(int playerIndex) throws Exception {
         } catch (BadCallerException e) {
             e.printStackTrace();
         }
-
         return turnTracker.nextTurn();
+
```
Comment (derekargueta) [inline java/src/shared/model/game/Game.java]: Please undo these changes

File: java/src/server/facade/MockFacade.java [hunk 2]
```diff
@@ -298,6 +298,7 @@ public CommandExecutionResult buildCity(int gameID, int player, VertexLocation l
      */
     @Override
     public CommandExecutionResult offerTrade(int gameID, OfferTradeDTO dto) throws OfferTradeException {
+
```
Comment (derekargueta) [inline java/src/server/facade/MockFacade.java]: Undo this

**Comment by derekargueta** [comment]
@CaptBoscho what is the purpose of this pull request? Please include a description.

File: java/src/shared/model/game/Game.java [hunk 3]
```diff
@@ -845,7 +869,9 @@ public void rollNumber(final int value) throws Exception {
             });
 
             //Move to next phase - Playing
-            turnTracker.nextPhase();
+            //turnTracker.nextPhase();
+            turnTracker.setPhase(TurnTracker.Phase.PLAYING);
+            System.out.println(turnTracker.getPhase().toString());
```
Comment (derekargueta) [inline java/src/shared/model/game/Game.java]: Delete this print-out

File: java/src/shared/model/game/Game.java [hunk 4]
```diff
@@ -845,7 +869,9 @@ public void rollNumber(final int value) throws Exception {
             });
 
             //Move to next phase - Playing
-            turnTracker.nextPhase();
+            //turnTracker.nextPhase();
```
Comment (derekargueta) [inline java/src/shared/model/game/Game.java]: Delete commented code

File: java/src/shared/model/game/TurnTracker.java [hunk 5]
```diff
@@ -82,6 +82,9 @@ public int nextTurn() throws Exception {
         } else {
             currentTurn++;
             currentTurn %= NUM_PLAYERS;
+            if(phase == Phase.PLAYING){
+                phase = Phase.ROLLING;
+            }
```
Comment (derekargueta) [inline java/src/shared/model/game/TurnTracker.java]: @yeyyosing @dannyharding10 please verify that this is desired behavior.

File: java/src/shared/model/game/Game.java [hunk 6]
```diff
@@ -813,6 +819,15 @@ public void discardCards(int playerIndex, List<ResourceType> cards) throws Playe
         if (canDiscardCards(playerIndex) && this.turnTracker.canDiscard()) {
             playerManager.discardResourceType(playerIndex, cards);
         }
+        boolean change = true;
```
Comment (derekargueta) [inline java/src/shared/model/game/Game.java]: Please use a more descriptive variable name than `change`. It is difficult to tell exactly what the purpose of this code is.

**Comment by CaptBoscho** [comment]
The purpose is in the game in initiate settlement to give players their
resources if it's the set up turn 2

File: java/src/shared/model/game/Game.java (hunk 6 above)
Comment (CaptBoscho) [inline java/src/shared/model/game/Game.java]: I did, so I made this pull request am expression pod cdebug, where those
changes were after Danny pointed it out. So that once first, sorry I
extended it

File: java/src/shared/model/game/Game.java [hunk 7]
```diff
@@ -826,13 +841,22 @@ public void rollNumber(final int value) throws Exception {
...
                 if (player.canDiscardCards()) {
-                    turnTracker.setPhase(TurnTracker.Phase.DISCARDING);
+                    player.setDiscarded(false);
+                    discard = true;
+                }else{
+                    player.setDiscarded(true);
                 }
-            });
+            }
+            if(discard){
```
Comment (derekargueta) [inline java/src/shared/model/game/Game.java]: This can be simplified with a ternary operator.

File: java/src/shared/model/game/TurnTracker.java (hunk 5 above)
Comment (CaptBoscho) [inline java/src/shared/model/game/TurnTracker.java]: He was with me when I did it

**Comment by derekargueta** [comment]
@dannyharding10 @yeyyosing please review
